### ISSUE TITLE
[BUG] Cannot find name 'Element' and 'Node'

### ISSUE THREAD

**Issue by thiagonsiq** [node-saml/passport-saml#691]
**To Reproduce**

Start of the node application triggers the following errors: 

node_modules/passport-saml/lib/node-saml/saml.d.ts:54:38 - error TS2304: Cannot find name 'Node'.
54     _getNameIdAsync(self: SAML, doc: Node): Promise<NameID>;
node_modules/passport-saml/lib/node-saml/saml.d.ts:54:38 - error 

TS2304: Cannot find name 'Node'.
54     _getNameIdAsync(self: SAML, doc: Node): Promise<NameID>;

**Expected behavior**

Expect to be able to npm run start and have the node application start as normal.

**Environment**

- Node.js version: 16.14.2
- `passport-saml` version: 3.2.1 (also tried with 3.2.0 and 3.1.2)
- `node-saml` version: 4.0.0-beta.2

**Comment by srd90** [comment]
@thiagonsiq you wrote that environment is:
>* `passport-saml` version: 3.2.1 (also tried with 3.2.0 and 3.1.2)
>* `node-saml` version: 4.0.0-beta.2

Just out of curiosity why are you mixing `passport-saml 3.2.1` and `node-saml 4.0.0-beta.2` packages into same application?

`node-saml` is (shall be) transitive dependecy of `passport-saml 4.x`. `passport-saml` was splitted to passport specific part and separated node-saml package which is does not depend on any specific framewor.

I.e. if you are using `passprot-saml` module you should let it decide which `node-saml` version it "pulls" as transitive dependency.

`passport-saml` versions `< 4.x` contain `node-saml` stuff "inline".

Starting from `passport-saml` `>= 4.x` (it is not yet officially released) it has `node-saml` as one of it dependencies and your dependency management software shall resolve proper `node-saml` version for you (i.e. you should have only `passport-saml` in you `package.json` file and if you do not need anything from `passport-saml` module leave it out and put `node-saml` to `package.json`).

`passport-saml 3.2.1`'s dependencies are
https://github.com/node-saml/passport-saml/blob/6ba76ba3a015fea96a2dd38f661a6c1f85bc44a1/package.json#L52-L60
(note there isn't `node-saml` because those libraries are still in `passport-saml`: https://github.com/node-saml/passport-saml/tree/6ba76ba3a015fea96a2dd38f661a6c1f85bc44a1/src ).

`passport-saml 4.x` i.e. current `master` which is not yet released has:
https://github.com/node-saml/passport-saml/blob/5fd8c401469e8b7e511b29abc5b3ff7434b0f039/package.json#L52-L56

---

@cjbarth @markstos is concept of transitive dependencies and dependency management somewhat "unknown" in nodejs ecosystem and if so should `passport-saml` documentation have some "compatibility matrix" at the start of readme?

There are also other projects which mix `passport-saml 3.x` and `node-saml 4.x` packages. It starts to look as if people are assuming that they should take latest `passport-saml` and latest `node-saml` and expect those to work together. Disclaimer: I do not know whether this issue's root cause is mixture but it is quite obvious that `passport-saml 3.2.1` and `node-saml 4.x` are not designed to work together.

Examples of other projects which mix these packages (links are pinned to current `HEAD`s of `master` branches of those projects):
* [perforce/helix-authentication-service](https://github.com/perforce/helix-authentication-service/tree/bedb84dbf6be8f5c7b9e1257d0997c7c110ad26a)
  * [package-json#L53-L58](https://github.com/perforce/helix-authentication-service/blob/bedb84dbf6be8f5c7b9e1257d0997c7c110ad26a/package.json#L53-L58)
* [ais-one/cookbook](https://github.com/ais-one/cookbook/tree/68bdc6e279e31f3ad6754b48294ac77b4f1d5a85)
  * [js-node/expressjs/package.json#L46-L48](https://github.com/ais-one/cookbook/blob/68bdc6e279e31f3ad6754b48294ac77b4f1d5a85/js-node/expressjs/package.json#L46-L48)
  * this project seems to be some collection of "template projects" i.e. projects which are used to "spawn" actual projects

FWIW: maintainers of aforementioned projects were "pinged" at https://github.com/node-saml/node-saml/issues/21 issue (which was actually about mixing `node-saml/passport-saml` and `zoellner/node-saml` modules) but it seems that they have choosed to continue to use untested combination (by "fixing" problem described at that issue so that now they use `passport-saml 3.x` and `node-saml 4.x`).

**Comment by thiagonsiq** [comment]
Unfortunately removing node-saml did not fix the problem, although I did it anyway since it does not make sense to have both on the project. 

The problem shows up when I import "passport-saml" into my project to be able to use "Strategy" 

import { Strategy } from 'passport-saml'

I also made sure all the dependencies were installed as stated above.
If I update the saml.d.ts file and change the "Element" and "Node" references to "any", everything works, but this isn't a good bandaid because whenever anyone does an "npm i", the problem returns.
